src/lib.rs

```diff
@@ -132,11 +137,22 @@ where
 ///     B(EventB),
 /// }
 ///
-/// impl EventData for DomainEvents {}
+/// impl EventData for DomainEvents {
+///   fn event_type(&self) -> String {
+///     match &self {
+///         DomainEvents::A(_) => "namespace.A",
+///         DomainEvents::B(_) => "namespace.B",
+///         _ => panic!("Woops"),
+///     }.into()
+///   }
+/// }
```

Comment:
After [my other comment](TODO), you can remove this from the doc comment as it's using the default impl on `EventData`.